### PR TITLE
fix: correct inverted null guard on client settings attachments

### LINKED ISSUE #161: B10: Inverted null guard in ReportPostOptionsExtensions
## Problem
`clientSettings.Attachments?.Count != 0` is `true` when `Attachments` is null, so `AddRange(null)` would throw (`Runtime/Util/ReportPostOptionsExtensions.cs`). Latent today — both settings repos initialize the list — but one refactor away from live.

## Fix
`?.Count > 0`. Add the unit test (#132 J2's empty stub is precisely the file that should have caught this).

Sliced from the 5.0.0 audit #132; full context and benchmarks there.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by bobbyg603**
Closes #161

## Problem

`Runtime/Util/ReportPostOptionsExtensions.cs` guarded the attachment copy with:

```csharp
if (clientSettings.Attachments?.Count != 0)
{
    options.AdditionalAttachments.AddRange(clientSettings.Attachments);
}
```

When `Attachments` is null, `?.Count` yields a null `int?` and `null != 0` is **true**, so the guard falls through into `AddRange(null)` and throws `ArgumentNullException`. The guard was exactly inverted: it skipped the copy only in the one case where copying is harmless (an empty list) and admitted it in the case that throws.

## Fix

```csharp
if (clientSettings.Attachments?.Count > 0)
```

`>` on a lifted `int?` is false when the operand is null, so a null `Attachments` correctly skips the copy.

## Verification of the "latent today" claim

Confirmed still latent on `main` — both shipped settings repositories initialize the list:

- `WebGLClientSettingsRepository.Attachments` is initialized inline (`= new List<FileInfo>()`).
- `DotNetStandardClientSettingsRepository.Attachments` delegates to `BugSplatDotNetStandard.BugSplat.Attachments`. I instantiated the vendored `Runtime/Plugins/BugSplatDotNetStandard.dll` directly and confirmed the property is non-null (empty list) on a fresh instance.

So nothing crashes today, but any new `IClientSettingsRepository` implementation that leaves the property null would throw on every report post.

## Other collections in the file

I checked the rest of `SetNullOrEmptyValues` and grepped `Runtime/` and `Editor/` for `?.Count` / `?.Length`. The inverted guard appeared exactly once — the line fixed here is the only instance in the repo.

One adjacent observation, left unchanged to keep this fix minimal: `foreach (var attribute in clientSettings.Attributes)` has no null guard at all, so it would `NullReferenceException` under the same "future repository leaves it null" scenario. Same latency argument (both repos initialize `Attributes`), different pattern from the inverted guard this issue describes. Happy to fold it in if a reviewer prefers.

## Tests

`Tests/Runtime/Util/ExceptionPostOptionsExtensionsTests.cs` was an empty TODO stub — precisely the file that should have caught this. Filled it with coverage scoped to the null-guard behaviour:

- `SetNullOrEmptyValues_NullAttachments_ShouldNotThrow`
- `SetNullOrEmptyValues_EmptyAttachments_ShouldNotAddAttachments`
- `SetNullOrEmptyValues_PopulatedAttachments_ShouldAddAttachments`
- `Attachments_ShippedClientSettingsRepository_ShouldBeInitialized` (regression guard on the "latent" premise)

The null test is a genuine regression test: reverted to `!= 0`, it fails with exactly `System.ArgumentNullException: Value cannot be null. (Parameter 'collection')` at `List.AddRange` / `ReportPostOptionsExtensions.cs:12`. Restored, it passes.

> [!NOTE]
> #167 is concurrently expanding this same file (it fills both empty stubs, `ShouldPostExceptionImplTests.cs` and `ExceptionPostOptionsExtensionsTests.cs`). I kept this change tightly scoped to the null-guard behaviour, but expect a small merge in `ExceptionPostOptionsExtensionsTests.cs`.

## How I verified

Unity's test runner was not available here, so I compiled and ran these types out-of-editor — they are plain C# with no Unity runtime dependency.

1. **Full compile**: adapted `pkgcheck.csproj` (whole `Runtime/**` + `Tests/Runtime/**` against the Unity managed DLLs and the vendored `BugSplatDotNetStandard.dll`) — `dotnet build` succeeds, **0 errors**, only the two pre-existing `CS0649` warnings in `BugSplat.cs`.
2. **Actually ran the tests**: a second project compiling the non-Unity-test-infra subset (`ReportPostOptions`, the settings repositories, `ReportPostOptionsExtensions`, `ShouldPostExceptionImpl`, `NativeSyncDictionary`, plus `ExceptionPostOptionsExtensionsTests` and `NativeSyncDictionaryTests`) with NUnit 3.14 + NUnit3TestAdapter + Microsoft.NET.Test.Sdk.

```
Passed!  - Failed: 0, Passed: 15, Skipped: 0, Total: 15
```

15 = my 4 new tests + the 11 existing `NativeSyncDictionaryTests`. The `[UnityTest]`-based suites need the Unity editor and were compile-checked only, not executed.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by daveplunkett — APPROVED** [review]

File: Tests/Runtime/Util/ExceptionPostOptionsExtensionsTests.cs [hunk 1]
```diff
@@ -1,13 +1,75 @@
...
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.IO;
 
 namespace BugSplatUnity.RuntimeTests.Util
 {
-    class ExceptionPostOptionsExtensionsTests
+    public class ExceptionPostOptionsExtensionsTests
```
[the remark this answers is not in the document]
Comment (bobbyg603) [reply Tests/Runtime/Util/ExceptionPostOptionsExtensionsTests.cs]: Fixed in c65734e — renamed the class and file to `ReportPostOptionsExtensionsTests` (the stub kept its original name from #31 even though there's no `ExceptionPostOptionsExtensions` class in the codebase). The `.meta` was renamed alongside via `git mv`, so the Unity GUID is preserved.

File: Tests/Runtime/Util/ReportPostOptionsExtensionsTests.cs [hunk 2]
```diff
@@ -1,13 +1,75 @@
...
+
+            Assert.AreEqual(1, options.AdditionalAttachments.Count);
+            Assert.AreSame(attachment, options.AdditionalAttachments[0]);
+        }
+
+        [Test]
+        public void Attachments_ShippedClientSettingsRepository_ShouldBeInitialized()
+        {
+            Assert.IsNotNull(new WebGLClientSettingsRepository().Attachments);
+        }
```
[the remark this answers is not in the document]
Comment (bobbyg603) [reply Tests/Runtime/Util/ReportPostOptionsExtensionsTests.cs:56]: Fixed in c65734e — went with the rename: the test is now `Attachments_WebGLClientSettingsRepository_ShouldBeInitialized`. Expanding to the other shipped repository isn't practical here: `DotNetStandardClientSettingsRepository.Attachments` is a passthrough to `BugSplatDotNetStandard.BugSplat.Attachments`, and the test asmdef sets `overrideReferences: true` with only `nunit.framework.dll`, so asserting it would mean adding `BugSplatDotNetStandard.dll` to the test assembly's references just to pin an external package's initialization contract. WebGL is the only shipped repository whose `Attachments` initialization is this repo's own code.
